Tests/LibGfx/TestImageDecoder.cpp

```diff
@@ -2075,6 +2075,17 @@ TEST_CASE(test_jxl_icc)
     verify_checkerboard(*plugin_decoder);
 }
 
+TEST_CASE(test_jxl_palette_and_groups)
+{
+    auto file = TRY_OR_FAIL(Core::MappedFile::map(TEST_INPUT("jxl/palette_and_groups.jxl"sv)));
+    EXPECT(Gfx::JPEGXLImageDecoderPlugin::sniff(file->bytes()));
+    auto plugin_decoder = TRY_OR_FAIL(Gfx::JPEGXLImageDecoderPlugin::create(file->bytes()));
+
+    TRY_OR_FAIL(expect_single_frame_of_size(*plugin_decoder, { 64, 138 }));
+    auto frame = TRY_OR_FAIL(plugin_decoder->frame(0));
+    EXPECT_EQ(frame.image->get_pixel(20, 130), Gfx::Color::from_string("#0b1112"sv));
```

Comment:
nit: `Gfx::Color(0x0b, 0x11, 0x12)` is shorter and doesn't drag string conversions into this.

Also, very nit: could compare all pixel to a decoded `png/buggie.png` (or is this a lossy image?)

Reply:
I prefer the hex-string format, it's a bit more readable and easier to compare with the output of other apps.

I guess that the PNG decoder is mature enough to rely on it for other tests, but I would rather keep all tests independent. I never took the time to do it, but I imagine something like having a decoded buggie as `#embed`ed bytes and then testing decoders against it.

Reply:
/shruggie

All my decoders compare against PNG references 😅